src/main/scala/scalaz/ziokeeper/VectorClock.scala

```diff
@@ -0,0 +1,64 @@
+package scalaz.ziokeeper
+
+import scala.annotation.tailrec
+
+case class VectorClock[A](value: Map[A, Long])
+
+object VectorClock {
+  def empty[A] = VectorClock[A](value = Map.empty)
+
+  implicit class VectorClockOps[A](val v: VectorClock[A]) extends AnyVal {
+    def time(mid: A): Long = v.value.getOrElse(mid, 0L)
+
+    def increment(mid: A, value: Long): VectorClock[A] =
+      VectorClock(v.value.updated(mid, time(mid) + value))
+  }
+
+  sealed trait CompareVectorClock
```

Comment:
I was trying to avoid ambiguity with scala.Ordering which is not the same and I need to implement Ordering anyway but I need to read one more paper to figure out what to do when I we have concurrent modifications of the same elements.

So do you have better name for that?

Reply:
I see. Not at the moment, let's leave it as it is, and change once we find the more suitable name.